src/components/organize/campaigns/CamapignCard.tsx

```diff
@@ -3,58 +3,44 @@ import { useRouter } from 'next/router';
 import { Card, CardActions, CardContent, Link, Typography } from '@material-ui/core';
 import { FormattedDate, FormattedMessage as Msg } from 'react-intl';
 
-import { getNaiveDate } from 'utils/dateUtils';
+import { removeOffset } from 'utils/dateUtils';
 import { ZetkinCampaign, ZetkinEvent } from 'types/zetkin';
 
 interface CampaignCardProps {
     campaign: ZetkinCampaign;
-    events: ZetkinEvent[];
-    upcomingEvents: ZetkinEvent[];
+    firstEvent?: ZetkinEvent;
+    lastEvent?: ZetkinEvent;
+    numOfUpcomingEvents?: number;
 }
 
-const CamapignCard = ({ campaign, events, upcomingEvents }: CampaignCardProps) : JSX.Element => {
+const CamapignCard = ({ campaign, firstEvent, lastEvent, numOfUpcomingEvents }: CampaignCardProps) : JSX.Element => {
     const { orgId } = useRouter().query;
     const { id, title } = campaign;
 
-    const campaignEvents = events.filter(e => e.campaign.id === id);
-    const campaignUpcomingEvents = upcomingEvents.filter(e => e.campaign?.id === id);
-
-    let endDate, startDate;
-
-    const firstEvent = campaignEvents[0];
-    const lastEvent = campaignEvents[campaignEvents.length - 1];
-    if (firstEvent && lastEvent) {
-        startDate = getNaiveDate(firstEvent.start_time) ;
-        endDate = getNaiveDate(lastEvent.end_time);
-    }
-
     return (
         <Card data-testid="campaign-card">
             <CardContent>
                 <Typography gutterBottom noWrap variant="h6">
                     { title }
                 </Typography>
                 <Typography gutterBottom variant="body2">
-                    { startDate && endDate ? (
+                    { firstEvent?.start_time && lastEvent?.end_time ? (
```

Comment:
This check makes more sense to me than the previous ones, but I may not understand entirely.